cpp/include/cuvs/core/c_api.h

```diff
@@ -127,6 +127,24 @@ cuvsError_t cuvsRMMAlloc(cuvsResources_t res, void** ptr, size_t bytes);
  */
 cuvsError_t cuvsRMMFree(cuvsResources_t res, void* ptr, size_t bytes);
 
+/**
+ * @brief Switches the memory resource to use the pool memory resource
+ *
+ * @param[in] initial_pool_size_percent The initial pool size as a percentage of the total
+ * available memory
+ * @param[in] max_pool_size_percent The maximum pool size as a percentage of the total
+ * available memory
+ * @return cuvsError_t
+ */
+cuvsError_t cuvsRMMEnablePoolMemoryResource(int initial_pool_size_percent,
```

Comment:
RMM's APIs also allow for a managed pool (a memory pool in managed memory) by accepting a boolean argument `managed`. I think we should add that argument here too.

Reply:
`cuvsRMMPoolMemoryResourceEnable` would be more consistent with our C-API naming elsewhere.

Reply:
> RMM's APIs also allow for a managed pool (a memory pool in managed memory) by accepting a boolean argument `managed`. I think we should add that argument here too.

I cannot find this argument in the [rmm documentation](https://docs.rapids.ai/api/rmm/stable/librmm_docs/memory_resources/#classrmm_1_1mr_1_1pool__memory__resource), but there is a seperate [managed_memory_resource](https://docs.rapids.ai/api/rmm/stable/librmm_docs/memory_resources/#classrmm_1_1mr_1_1managed__memory__resource). Should I add this as well?

Reply:
You are right, they changed the APIs more recently so they no longer include the “managed” argument in the pool resource API. However, they can get away with this in the c++ layer since the memory resources are compostable… we can’t get away with this in the C layer, unfortunately, and so I’d prefer to still provide the “managed” argument and smartly wrap a ManagedMemoryResource around the pool memory resource when it is set to true.